integration-tests/contract-originate-contract-with-sapling.spec.ts

```diff
@@ -43,5 +61,132 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
 
       done();
     });
+
+    test('Originates a Sapling Double contract', async (done) => {
+      const op = await Tezos.contract.originate({
+        code: saplingContractDouble,
+        init: `(Pair {} {})`,
+      });
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      const contract = await op.contract();
+
+      const saplingDiffById = await rpcClient.getSaplingDiffById('168');
+      expect(saplingDiffById).toBeDefined();
+
+      Tezos.contract.at(contract.address).then((contract) => {
+        const objects = Object.keys(contract.methodsObject);
+        expect(objects).toContain('default');
+      });
+
+      const storage: StorageType = await contract.storage();
+      const saplingStateLedger1 = await storage.left.getSaplingDiff();
+
+      expect(saplingStateLedger1.root).toBeDefined();
+      expect(saplingStateLedger1.nullifiers.length).toEqual(0);
+      expect(saplingStateLedger1.commitments_and_ciphertexts.length).toEqual(0);
+
+      done();
+    });
+
+    test('Originates a Sapling Drop contract', async (done) => {
+      const op = await Tezos.contract.originate({
+        code: saplingContractDrop,
+        init: { prim: 'Unit' },
+      });
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      const contract = await op.contract();
+
+      Tezos.contract.at(contract.address).then((contract) => {
+        const objects = Object.keys(contract.methodsObject);
+        expect(objects).toContain('default');
+      });
+      done();
+    });
+
+    test('Originates a Sapling Send contract', async (done) => {
+      const op = await Tezos.contract.originate({
+        code: saplingContractSend,
+        init: { prim: 'Unit' },
+      });
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      const contract = await op.contract();
+
+      Tezos.contract.at(contract.address).then((contract) => {
+        const objects = Object.keys(contract.methodsObject);
+        expect(objects).toContain('default');
+      });
+      done();
+    });
+
+    test('Originates a Sapling Contract State as Arg contract', async (done) => {
+      const op = await Tezos.contract.originate({
+        code: saplingContractStateAsArg,
+        storage: null,
+      });
+
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+      done();
+    });
+
+    hangzhounet('Should fail to originate a Push Sapling State contract', async (done) => {
```

Comment:
This test and the following one could be combined into one by matching the error message only to `michelson_v1.unexpected_lazy_storage`

Reply:
changed to be one test. Thanks that is much better :)